src/Tools/dotnet-dump/Dumper.cs

```diff
@@ -13,47 +14,61 @@ public Dumper()
         {
         }
 
-        public async Task<int> Collect(IConsole console, int processId, string outputDirectory)
+        public async Task<int> Collect(IConsole console, int processId, string output, string type)
```

Comment:
Just curious, System.CommandLine doesn't do enums?

Reply:
I did't even try. I see if a enum works.

Reply:
it works.

Reply:
Another thing that works in System.CommandLine is complex objects, so for example you could do this:

```cs
public class CollectOptions
{
    public CollectOptions(string output) => Output = output; // binding will set constructors
    int ProcessId { get ; set; } // binding will set setters
    string Type { get; set; }
}
// ... 
public async Task<int> Collect(IConsole console, CollectOptions options)
```

It's also worth noting that it will bind `FileInfo` and `DirectoryInfo` (and any other type having a constructor that takes a single string), so you don't need to use strings in your handler methods if you prefer not to.

And in case it's not clear, the IConsole and other dependency-injected service types can be placed on the CollectionOptions if you prefer.